src/compiler/checker.ts

```diff
@@ -14613,8 +14536,10 @@ namespace ts {
                 emitAwaiter = false;
                 potentialThisCollisions.length = 0;
 
+                deferredNodes = [];

```

Comment:
I can't see what's above this because it's above the 11K mark, but does that mean that this is unconditionally initialized? If so, you can remove the initialization check in `checkNodeDeferred`.